src-tauri/src/binaries/adapter_github.rs

```diff
@@ -0,0 +1,106 @@
+use log::{error, info};
+use regex::Regex;
+use tauri::api::path::cache_dir;
+
+use crate::{
+    download_utils::download_file_with_retries, github, progress_tracker::ProgressTracker,
+};
+
+use super::binaries_resolver::{
+    LatestVersionApiAdapter, VersionAsset, VersionDownloadInfo, BINARY_RESOLVER_LOG_TARGET,
+};
+
+pub struct GithubReleasesAdapter {
+    pub repo: String,
+    pub owner: String,
+    pub specific_name: Option<Regex>,
+}
+
+#[async_trait]
+impl LatestVersionApiAdapter for GithubReleasesAdapter {
+    async fn fetch_releases_list(&self) -> Result<Vec<VersionDownloadInfo>, Error> {
+        let releases = github::list_releases(&self.owner, &self.repo).await?;
+        Ok(releases.clone())
+    }
+
+    async fn download_and_get_checksum_path(
+        &self,
+        directory: PathBuf,
+        download_info: VersionDownloadInfo,
+        progress_tracker: ProgressTracker,
+    ) -> Result<PathBuf, Error> {
+        let asset = self.find_version_for_platform(&download_info)?;
+        let checksum_path = directory
+            .join("in_progress")
+            .join(format!("{}.sha256", asset.name));
+        let checksum_url = format!("{}.sha256", asset.url);
+
+        match download_file_with_retries(&checksum_url, &checksum_path, progress_tracker).await {
+            Ok(_) => Ok(checksum_path),
+            Err(e) => {
+                error!(target: BINARY_RESOLVER_LOG_TARGET, "Failed to download checksum file: {}", e);
+                Err(e)
+            }
+        }
+    }
+
+    fn get_binary_folder(&self) -> PathBuf {
+        let binary_folder_path = cache_dir()
+            .unwrap()
+            .join("com.tari.universe")
+            .join(&self.repo);
+
+        if !binary_folder_path.exists() {
+            drop(std::fs::create_dir_all(&binary_folder_path));
+        }
+
+        binary_folder_path
+    }
+
+    fn find_version_for_platform(
+        &self,
+        version: &VersionDownloadInfo,
+    ) -> Result<VersionAsset, Error> {
+        let mut name_suffix = "";
+        // TODO: add platform specific logic
+        if cfg!(target_os = "windows") {
+            name_suffix = r"windows-x64.*\.zip";
+        }
+
+        if cfg!(target_os = "macos") && cfg!(target_arch = "x86_64") {
+            name_suffix = r"macos-x86_64.*\.zip";
+        }
+
+        if cfg!(target_os = "macos") && cfg!(target_arch = "aarch64") {
+            name_suffix = r"macos-arm64.*\.zip";
+        }
+        if cfg!(target_os = "linux") {
+            name_suffix = r"linux-x86_64.*\.zip";
+        }
+        if name_suffix.is_empty() {
+            panic!("Unsupported OS");
+        }
+
+        info!(target: BINARY_RESOLVER_LOG_TARGET, "Looking for platform with suffix: {}", name_suffix);
+
+        let reg = Regex::new(name_suffix).unwrap();
+
+        let platform = version
+            .assets
+            .iter()
+            .find(|a| {
+                if let Some(ref specific) = self.specific_name {
```

Comment:
Please check the latest logic for this

Reply:
From what I see it's the same as on main branch